federation/pkg/federation-controller/service/servicecontroller.go

```diff
@@ -565,6 +565,8 @@ func getOperationsToPerformOnCluster(informer fedutil.FederatedInformer, cluster
 				}
 			}
 		}
+		// ExternalIPs are not managed by Kubernetes, so retain the same if any while updating
+		desiredService.Spec.ExternalIPs = clusterService.Spec.ExternalIPs
```

Comment:
I am not sure if this is ideal. I see that you are doing this for existing cluster services that are adopted. What if different service shards in different clusters have different external IPs?

Reply:
I'm not sure if I understand you correctly. I'm doing the same what is done few lines above with a ClusterIP and NodePort which are specific to a cluster service.
I don't touch federated service.

Reply:
ClusterIP and NodePort are automatically assigned by the underlying cluster's service controller.  ExternalIP is not.  So I don't think that what you are doing is correct.  It looks like what you're wanting to do is manually assign a different ExternalIP in each of the underlying cluster's service specs, and have that persist.  That is not consistent with the general approach of having the federated service spec be the source of truth.

Can you explain exactly what you're trying to achieve here.  Perhaps we can find a good way to support it.  I don't think that we want to merge this.

Reply:
We have a tool(https://github.com/Mirantis/k8s-externalipcontroller) which assigns ExternalIP to a service.
IP is deleted by federated controller when it updates Service. Each underlying cluster's service needs different ExternalIP.
In my opinion setting ExternalIP for federated service doesn't make sense. How would you configure routes for it?

Reply:
@lukaszo Yes, you have a good point.  And sorry for the delay on this one.

Please open a PR against https://github.com/kubernetes/federation for this change, cross-referencing the current PR, and I will merge it there.  Federation code has moved to https://github.com/kubernetes/federation.